In [1]:
import pandas as pd

롤 랭킹 데이터 : https://www.kaggle.com/datasnaek/league-of-legends


DataUrl = ‘https://raw.githubusercontent.com/Datamanim/pandas/main/lol.csv’

In [4]:
DataUrl = 'https://raw.githubusercontent.com/Datamanim/pandas/main/lol.csv'


df = pd.read_csv(DataUrl, sep='\t')

In [5]:
df

gameId   creationTime  gameDuration  seasonId  winner  firstBlood  \
0      3326086514  1504279457970          1949         9       1           2   
1      3229566029  1497848803862          1851         9       1           1   
2      3327363504  1504360103310          1493         9       1           2   
3      3326856598  1504348503996          1758         9       1           1   
4      3330080762  1504554410899          2094         9       1           2   
...           ...            ...           ...       ...     ...         ...   
51485  3308904636  1503076540231          1944         9       2           1   
51486  3215685759  1496957179355          3304         9       2           1   
51487  3322765040  1504029863961          2156         9       2           2   
51488  3256675373  1499562036246          1475         9       2           2   
51489  3317333020  1503612754059          1445         9       1           1   

       firstTower  firstInhibitor  firstBaron  firstDragon  ...  \
0               1               1           1            1  ...   
1               1               1           0            1  ...   
2               1               1           1            2  ...   
3               1               1           1            1  ...   
4               1               1           1            1  ...   
...           ...             ...         ...          ...  ...   
51485           2               2           0            2  ...   
51486           1               2           2            2  ...   
51487           2               2           0            1  ...   
51488           2               2           0            2  ...   
51489           1               1           1            2  ...   

       t2_towerKills  t2_inhibitorKills  t2_baronKills  t2_dragonKills  \
0                  5                  0              0               1   
1                  2                  0              0               0   
2                  2                  0              0               1   
3                  0                  0              0               0   
4                  3                  0              0               1   
...              ...                ...            ...             ...   
51485             10                  2              0               4   
51486             11                  7              4               4   
51487             10                  2              0               2   
51488             11                  3              0               1   
51489              1                  0              0               1   

       t2_riftHeraldKills  t2_ban1  t2_ban2  t2_ban3  t2_ban4  t2_ban5  
0                       1      114       67       43       16       51  
1                       0       11       67      238       51      420  
2                       0      157      238      121       57       28  
3                       0      164       18      141       40       51  
4                       0       86       11      201      122       18  
...                   ...      ...      ...      ...      ...      ...  
51485                   0       55       -1       90      238      157  
51486                   1      157       55      119      154      105  
51487                   0      113      122       53       11      157  
51488                   0      154       39       51       90      114  
51489                   0       11      157      141       31       18  

[51490 rows x 61 columns]

데이터의 상위 5개 행을 출력하라

In [6]:
df.head()

gameId   creationTime  gameDuration  seasonId  winner  firstBlood  \
0  3326086514  1504279457970          1949         9       1           2   
1  3229566029  1497848803862          1851         9       1           1   
2  3327363504  1504360103310          1493         9       1           2   
3  3326856598  1504348503996          1758         9       1           1   
4  3330080762  1504554410899          2094         9       1           2   

   firstTower  firstInhibitor  firstBaron  firstDragon  ...  t2_towerKills  \
0           1               1           1            1  ...              5   
1           1               1           0            1  ...              2   
2           1               1           1            2  ...              2   
3           1               1           1            1  ...              0   
4           1               1           1            1  ...              3   

   t2_inhibitorKills  t2_baronKills  t2_dragonKills  t2_riftHeraldKills  \
0                  0              0               1                   1   
1                  0              0               0                   0   
2                  0              0               1                   0   
3                  0              0               0                   0   
4                  0              0               1                   0   

   t2_ban1  t2_ban2  t2_ban3  t2_ban4  t2_ban5  
0      114       67       43       16       51  
1       11       67      238       51      420  
2      157      238      121       57       28  
3      164       18      141       40       51  
4       86       11      201      122       18  

[5 rows x 61 columns]

데이터의 행과 열의 갯수를 파악하라

In [7]:
df.shape

(51490, 61)

전체 컬럼을 출력하라

In [8]:
df.columns

Index(['gameId', 'creationTime', 'gameDuration', 'seasonId', 'winner',
       'firstBlood', 'firstTower', 'firstInhibitor', 'firstBaron',
       'firstDragon', 'firstRiftHerald', 't1_champ1id', 't1_champ1_sum1',
       't1_champ1_sum2', 't1_champ2id', 't1_champ2_sum1', 't1_champ2_sum2',
       't1_champ3id', 't1_champ3_sum1', 't1_champ3_sum2', 't1_champ4id',
       't1_champ4_sum1', 't1_champ4_sum2', 't1_champ5id', 't1_champ5_sum1',
       't1_champ5_sum2', 't1_towerKills', 't1_inhibitorKills', 't1_baronKills',
       't1_dragonKills', 't1_riftHeraldKills', 't1_ban1', 't1_ban2', 't1_ban3',
       't1_ban4', 't1_ban5', 't2_champ1id', 't2_champ1_sum1', 't2_champ1_sum2',
       't2_champ2id', 't2_champ2_sum1', 't2_champ2_sum2', 't2_champ3id',
       't2_champ3_sum1', 't2_champ3_sum2', 't2_champ4id', 't2_champ4_sum1',
       't2_champ4_sum2', 't2_champ5id', 't2_champ5_sum1', 't2_champ5_sum2',
       't2_towerKills', 't2_inhibitorKills', 't2_baronKills', 't2_dragonKills',
       't2_riftHer

6번째 컬럼명을 출력하라

In [10]:
df.columns[5]

'firstBlood'

6번째 컬럼의 데이터 타입을 확인하라

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51490 entries, 0 to 51489
Data columns (total 61 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   gameId              51490 non-null  int64
 1   creationTime        51490 non-null  int64
 2   gameDuration        51490 non-null  int64
 3   seasonId            51490 non-null  int64
 4   winner              51490 non-null  int64
 5   firstBlood          51490 non-null  int64
 6   firstTower          51490 non-null  int64
 7   firstInhibitor      51490 non-null  int64
 8   firstBaron          51490 non-null  int64
 9   firstDragon         51490 non-null  int64
 10  firstRiftHerald     51490 non-null  int64
 11  t1_champ1id         51490 non-null  int64
 12  t1_champ1_sum1      51490 non-null  int64
 13  t1_champ1_sum2      51490 non-null  int64
 14  t1_champ2id         51490 non-null  int64
 15  t1_champ2_sum1      51490 non-null  int64
 16  t1_champ2_sum2      51490 non-null  int6

In [13]:
df['firstBlood']

0        2
1        1
2        2
3        1
4        2
        ..
51485    1
51486    1
51487    2
51488    2
51489    1
Name: firstBlood, Length: 51490, dtype: int64

In [14]:
df['firstBlood'].dtype

dtype('int64')

6번째 컬럼의 3번째 값은 무엇인가?

In [16]:
df['firstBlood'][2]

2

제주 날씨,인구에 따른 교통량데이터 : 출처 제주 데이터 허브 


DataUrl = ‘https://raw.githubusercontent.com/Datamanim/pandas/main/Jeju.csv’

In [19]:
DataUrl = 'https://raw.githubusercontent.com/Datamanim/pandas/main/Jeju.csv'

df = pd.read_csv(DataUrl,encoding='euc-kr')

In [20]:
df

id          일자   시도명  읍면동명         거주인구       근무인구        방문인구  \
0     22448  2018-01-01   제주시   도두동    32249.987   3418.266  102709.092   
1     22449  2018-01-01   제주시   외도동   213500.997  10341.172  112692.789   
2     22450  2018-01-01   제주시  이도2동  1212382.218  96920.834  541194.481   
3     22451  2018-01-01   제주시  일도1동    33991.653   6034.253   72155.919   
4     22452  2018-01-01  서귀포시   대천동   155036.925   9403.969  150882.409   
...     ...         ...   ...   ...          ...        ...         ...   
9616  32064  2020-04-30  서귀포시   표선면   228260.005  16891.732  152832.449   
9617  32065  2020-04-30   제주시   한림읍   459959.064  26007.122  249700.419   
9618  32066  2020-04-30   제주시   도두동    28397.481   3144.895   84052.697   
9619  32067  2020-04-30  서귀포시   안덕면   348037.846  29106.286  251129.660   
9620  32068  2020-04-30   제주시    연동  1010643.372  65673.477  447622.068   

           총 유동인구   평균 속도  평균 소요 시간  평균 기온  일강수량  평균 풍속  
0      138377.345  39.556    29.167    5.0   0.0   2.50  
1      336534.958  32.900    30.900    5.0   0.0   2.50  
2     1850497.533  29.538    35.692    2.9   0.0   2.40  
3      112181.825  30.000    23.500    2.9   0.0   2.40  
4      315323.303  41.583    14.375    5.1   0.0   2.30  
...           ...     ...       ...    ...   ...    ...  
9616   397984.186  44.514    34.054   18.1   0.0   4.70  
9617   735666.605  48.609    61.377   18.0   0.0   4.35  
9618   115595.073  41.053    29.421   20.3   0.0   3.00  
9619   628273.792  46.595    49.189   17.6   0.0   3.50  
9620  1523938.917  40.863    27.765   14.1   0.0   4.80  

[9621 rows x 13 columns]

데이터 마지막 3개행을 출력하라

In [21]:
df.tail(3)

id          일자   시도명 읍면동명         거주인구       근무인구        방문인구  \
9618  32066  2020-04-30   제주시  도두동    28397.481   3144.895   84052.697   
9619  32067  2020-04-30  서귀포시  안덕면   348037.846  29106.286  251129.660   
9620  32068  2020-04-30   제주시   연동  1010643.372  65673.477  447622.068   

           총 유동인구   평균 속도  평균 소요 시간  평균 기온  일강수량  평균 풍속  
9618   115595.073  41.053    29.421   20.3   0.0    3.0  
9619   628273.792  46.595    49.189   17.6   0.0    3.5  
9620  1523938.917  40.863    27.765   14.1   0.0    4.8

In [22]:
df.head(3)

id          일자  시도명  읍면동명         거주인구       근무인구        방문인구  \
0  22448  2018-01-01  제주시   도두동    32249.987   3418.266  102709.092   
1  22449  2018-01-01  제주시   외도동   213500.997  10341.172  112692.789   
2  22450  2018-01-01  제주시  이도2동  1212382.218  96920.834  541194.481   

        총 유동인구   평균 속도  평균 소요 시간  평균 기온  일강수량  평균 풍속  
0   138377.345  39.556    29.167    5.0   0.0    2.5  
1   336534.958  32.900    30.900    5.0   0.0    2.5  
2  1850497.533  29.538    35.692    2.9   0.0    2.4

In [23]:
df.shape

(9621, 13)

In [24]:
# 실제로 날짜는 몇일치의 데이터를 다루고 있나??

In [29]:
df['일자'].nunique()

803

In [30]:
df['일자'].unique()

array(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
       '2018-01-05', '2018-01-06', '2018-01-07', '2018-01-10',
       '2018-01-12', '2018-01-13', '2018-01-14', '2018-01-15',
       '2018-01-16', '2018-01-17', '2018-01-18', '2018-01-19',
       '2018-01-20', '2018-01-21', '2018-01-22', '2018-01-23',
       '2018-01-24', '2018-01-25', '2018-01-26', '2018-01-27',
       '2018-01-28', '2018-01-29', '2018-01-30', '2018-01-31',
       '2018-02-01', '2018-02-02', '2018-02-03', '2018-02-04',
       '2018-02-07', '2018-02-08', '2018-02-09', '2018-02-10',
       '2018-02-11', '2018-02-12', '2018-02-16', '2018-02-17',
       '2018-02-18', '2018-02-19', '2018-02-20', '2018-02-21',
       '2018-02-22', '2018-02-23', '2018-02-24', '2018-02-25',
       '2018-02-26', '2018-02-27', '2018-02-28', '2018-03-01',
       '2018-03-02', '2018-03-03', '2018-03-04', '2018-03-05',
       '2018-03-06', '2018-03-07', '2018-03-08', '2018-03-10',
       '2018-03-11', '2018-03-12', '2018-03-13', '2018-

In [25]:
# 이 데이터의 시도명도 중복사용되고 있다. 시도는 몇개가 있나??

In [33]:
df['시도명'].nunique()

2

In [36]:
df['시도명'].unique()

array(['제주시', '서귀포시'], dtype=object)

In [26]:
# 동은 몇개인가??

In [39]:
df['읍면동명'].nunique()

41

각 컬럼의 결측치 숫자를 파악하라

In [42]:
df.isna().sum()

id          0
일자          0
시도명         0
읍면동명        0
거주인구        0
근무인구        0
방문인구        0
총 유동인구      0
평균 속도       0
평균 소요 시간    0
평균 기온       0
일강수량        0
평균 풍속       0
dtype: int64

각 컬럼의 데이터수, 데이터타입을 한번에 확인하라

In [44]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9621 entries, 0 to 9620
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        9621 non-null   int64  
 1   일자        9621 non-null   object 
 2   시도명       9621 non-null   object 
 3   읍면동명      9621 non-null   object 
 4   거주인구      9621 non-null   float64
 5   근무인구      9621 non-null   float64
 6   방문인구      9621 non-null   float64
 7   총 유동인구    9621 non-null   float64
 8   평균 속도     9621 non-null   float64
 9   평균 소요 시간  9621 non-null   float64
 10  평균 기온     9621 non-null   float64
 11  일강수량      9621 non-null   float64
 12  평균 풍속     9621 non-null   float64
dtypes: float64(9), int64(1), object(3)
memory usage: 977.3+ KB


각 수치형 변수의 분포(사분위, 평균, 표준편차, 최대 , 최소)를 확인하라

In [45]:
df.describe()

id          거주인구           근무인구           방문인구        총 유동인구  \
count   9621.000000  9.621000e+03    9621.000000    9621.000000  9.621000e+03   
mean   27258.000000  3.174315e+05   35471.201510  195889.561802  5.487922e+05   
std     2777.487804  2.982079e+05   40381.214775  140706.090325  4.608802e+05   
min    22448.000000  9.305552e+03    1407.936000   11538.322000  2.225181e+04   
25%    24853.000000  9.539939e+04   12074.498000   99632.153000  2.216910e+05   
50%    27258.000000  2.221105e+05   21960.928000  152805.335000  3.866935e+05   
75%    29663.000000  4.106671e+05   40192.032000  236325.109000  6.406918e+05   
max    32068.000000  1.364504e+06  263476.965000  723459.209000  2.066484e+06   

             평균 속도     평균 소요 시간        평균 기온         일강수량        평균 풍속  
count  9621.000000  9621.000000  9621.000000  9621.000000  9621.000000  
mean     41.109084    37.215873    13.550828     6.972426     2.753171  
std       8.758631    12.993786     7.745515    27.617260     1.498538  
min      24.333000    12.667000    -9.600000     0.000000     0.000000  
25%      34.250000    27.889000     7.600000     0.000000     1.700000  
50%      39.640000    34.500000    13.400000     0.000000     2.400000  
75%      49.105000    46.176000    19.700000     1.500000     3.400000  
max     103.000000   172.200000    30.400000   587.500000    13.333000

거주인구 컬럼의 값들을 출력하라

In [47]:
df['거주인구']

0         32249.987
1        213500.997
2       1212382.218
3         33991.653
4        155036.925
           ...     
9616     228260.005
9617     459959.064
9618      28397.481
9619     348037.846
9620    1010643.372
Name: 거주인구, Length: 9621, dtype: float64

읍면동명 컬럼의 유일값 갯수를 출력하라

In [49]:
df['읍면동명'].nunique()

41

읍면동명 컬럼의 유일값을 모두 출력하라

In [50]:
df['읍면동명'].unique()

array(['도두동', '외도동', '이도2동', '일도1동', '대천동', '서홍동', '한경면', '송산동', '조천읍',
       '일도2동', '영천동', '예래동', '대륜동', '삼도1동', '이호동', '건입동', '중앙동', '삼양동',
       '삼도2동', '이도1동', '남원읍', '대정읍', '정방동', '효돈동', '아라동', '한림읍', '구좌읍',
       '용담1동', '오라동', '화북동', '연동', '표선면', '중문동', '성산읍', '안덕면', '천지동',
       '노형동', '동홍동', '용담2동', '봉개동', '애월읍'], dtype=object)

In [52]:
# 각 동별로, 평균 유동인구수를 구하고, 상위 5개동만 화면에 나타내시오.

In [57]:
df.groupby('읍면동명')['총 유동인구'].mean().sort_values(ascending=False).head()

읍면동명
이도2동    1.854812e+06
노형동     1.846636e+06
연동      1.729008e+06
애월읍     1.349167e+06
아라동     1.290003e+06
Name: 총 유동인구, dtype: float64

In [59]:
1.854812*10**6

1854812.0

식당데이터 : https://github.com/justmarkham/DAT8/blob/master/data/chipotle.tsv


DataUrl = ‘https://raw.githubusercontent.com/Datamanim/pandas/main/chipo.csv’

In [2]:
import pandas as pd

In [3]:
DataUrl = 'https://raw.githubusercontent.com/Datamanim/pandas/main/chipo.csv'
df = pd.read_csv(DataUrl)

In [4]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

quantity컬럼 값이 3인 데이터를 추출하여 첫 5행을 출력하라

In [5]:
df['quantity'] == 3

0       False
1       False
2       False
3       False
4       False
        ...  
4617    False
4618    False
4619    False
4620    False
4621    False
Name: quantity, Length: 4622, dtype: bool

In [6]:
df.loc[ df['quantity'] == 3 , ].head()

order_id  quantity          item_name  \
409       178         3       Chicken Bowl   
445       193         3               Bowl   
689       284         3  Canned Soft Drink   
818       338         3      Bottled Water   
850       350         3  Canned Soft Drink   

                                    choice_description item_price  
409  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...    $32.94   
445  [Braised Carnitas, Pinto Beans, [Sour Cream, C...    $22.20   
689                                        [Diet Coke]     $3.75   
818                                                NaN     $3.27   
850                                           [Sprite]     $3.75

quantity컬럼 값이 3인 데이터를 추출하여 index를 0부터 정렬하고 첫 5행을 출력하라

In [7]:
df.loc[ df['quantity']  ==  3 ,  ].reset_index().head()

index  order_id  quantity          item_name  \
0    409       178         3       Chicken Bowl   
1    445       193         3               Bowl   
2    689       284         3  Canned Soft Drink   
3    818       338         3      Bottled Water   
4    850       350         3  Canned Soft Drink   

                                  choice_description item_price  
0  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...    $32.94   
1  [Braised Carnitas, Pinto Beans, [Sour Cream, C...    $22.20   
2                                        [Diet Coke]     $3.75   
3                                                NaN     $3.27   
4                                           [Sprite]     $3.75

quantity , item_price 두개의 컬럼으로 구성된 새로운 데이터 프레임을 만드시오

In [8]:
df[ ['quantity' , 'item_price']  ]

quantity item_price
0            1     $2.39 
1            1     $3.39 
2            1     $3.39 
3            1     $2.39 
4            2    $16.98 
...        ...        ...
4617         1    $11.75 
4618         1    $11.75 
4619         1    $11.25 
4620         1     $8.75 
4621         1     $8.75 

[4622 rows x 2 columns]

item_price 컬럼의 달러표시 문자를 제거하고 float 타입으로 저장하여 new_price 컬럼에 저장하라

In [9]:
price = '30,000,000'

In [10]:
float( price.replace(',' , '') )

30000000.0

In [11]:
df['new_price'] = df['item_price'].str.replace('$', '' , regex = False).astype(float)

In [12]:
df.describe()

order_id     quantity    new_price
count  4622.000000  4622.000000  4622.000000
mean    927.254868     1.075725     7.464336
std     528.890796     0.410186     4.245557
min       1.000000     1.000000     1.090000
25%     477.250000     1.000000     3.390000
50%     926.000000     1.000000     8.750000
75%    1393.000000     1.000000     9.250000
max    1834.000000    15.000000    44.250000

new_price 컬럼이 5이하의 값을 가지는 데이터프레임을 추출하고, 데이터의 갯수는 몇개인가

In [13]:
df['new_price'] <= 5

0        True
1        True
2        True
3        True
4       False
        ...  
4617    False
4618    False
4619    False
4620    False
4621    False
Name: new_price, Length: 4622, dtype: bool

In [14]:
df2 = df.loc[  df['new_price'] <= 5 , ]

In [15]:
df2

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
...        ...       ...                                    ...   
4606      1828         1                      Canned Soft Drink   
4609      1829         1                      Canned Soft Drink   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4616      1832         1                    Chips and Guacamole   

     choice_description item_price  new_price  
0                   NaN     $2.39        2.39  
1          [Clementine]     $3.39        3.39  
2               [Apple]     $3.39        3.39  
3                   NaN     $2.39        2.39  
6                   NaN     $1.69        1.69  
...                 ...        ...        ...  
4606             [Coke]     $1.25        1.25  
4609           [Sprite]     $1.25        1.25  
4613                NaN     $2.15        2.15  
4614                NaN     $1.50        1.50  
4616                NaN     $4.45        4.45  

[1652 rows x 6 columns]

In [16]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  new_price  
0                                                   NaN     $2.39        2.39  
1                                          [Clementine]     $3.39        3.39  
2                                               [Apple]     $3.39        3.39  
3                                                   NaN     $2.39        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98       16.98  
...                                                 ...        ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[4622 rows x 6 columns]

item_name명이 Chicken Salad Bowl 인 데이터 프레임을 추출하고, index 값을 초기화 하여라

new_price값이 9 이하이고 item_name 값이 Chicken Salad Bowl 인 데이터 프레임을 추출하라

df의 new_price 컬럼 값에 따라 오름차순으로 정리하고 index를 초기화 하여라



df의 item_name 컬럼 값중 Chips 포함하는 경우의 데이터를 출력하라

df의 짝수번째 컬럼만을 포함하는 데이터프레임을 출력하라

df의 new_price 컬럼 값에 따라 내림차순으로 정리하고 index를 초기화 하여라

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 가져오시오

df의 데이터 중 new_price값이 new_price값의 평균값 이상을 가지는 데이터들을 가져오시오

df의 데이터 중 item_name의 값이 Izze 데이터를 Fizzy Lizzy로 수정하라

df의 데이터 중 choice_description 값이 NaN 인 데이터의 갯수를 구하여라

df의 데이터 중 choice_description 값이 NaN 인 데이터를 NoData 값으로 대체하라 

df의 데이터 중 choice_description 값에 Black이 들어가는 경우를 가져오시오

df의 데이터 중 choice_description 값에 Vegetables 들어가지 않는 경우의 갯수를 출력하라

df의 데이터 중 item_name 값이 N으로 시작하는 데이터를 모두 추출하라

df의 데이터 중 item_name 값의 단어갯수가 15개 이상인 데이터를 인덱싱하라

df의 데이터 중 new_price값이 lst에 해당하는 경우의 데이터 프레임을 구하고 그 갯수를 출력하라 

lst =[1.69, 2.39, 3.39, 4.45, 9.25, 10.98, 11.75, 16.98]

뉴욕 airBnB : https://www.kaggle.com/ptoscano230382/air-bnb-ny-2019 
        
DataUrl = ‘https://raw.githubusercontent.com/Datamanim/pandas/main/AB_NYC_2019.csv’

In [8]:
df= pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/AB_NYC_2019.csv')

데이터의 각 host_name의 빈도수를 구하고 host_name으로 정렬하여 상위 5개를 출력하라

데이터의 각 host_name의 갯수를 구하고 갯수 기준 내림차순 정렬한 데이터 프레임을 만들어라.  

neighbourhood_group의 값에 따른 neighbourhood컬럼 값의 갯수를 구하여라

neighbourhood_group의 값에 따른 neighbourhood컬럼 값 중 neighbourhood_group그룹의 최댓값들을 출력하라

neighbourhood_group 값에 따른 price값의 평균, 분산, 최대, 최소 값을 구하여라